Weather App

felixDev22 opened the pull request:
Hi Sir/Madam ✋ ;

In this repo:
 
- [ ] Used React components.
- [ ]  Used React props.
- [ ]  Used React Router.
- [ ]  Connected React and Redux.
- [ ]  Handle events in a React app.
- [ ]  Write integration tests with a React testing library.
- [ ]  Added styling for the home & details
- [ ]  Applied React best practices and language style guides in code.
- [ ]  Used store, actions, and reducers in React.
- [ ]  Fix the Linter's errors
- [ ]  Updated readme with professional documentation
- [ ]  Checked that the app is working properly

felixDev22 committed:
Add linters config files

felixDev22 committed:
Remove CRA boiler plate files

felixDev22 committed:
add global settings, add countryApi & weatherApi

felixDev22 committed:
Add weaher url fetch, createSlice

felixDev22 committed:
Fetch countryApI reducers, createSlice

felixDev22 committed:
Add the reducers in the configstore

felixDev22 committed:
Add Navbar, Homepage and  Details files

felixDev22 committed:
Add route path for the app files

felixDev22 committed:
Update index.js. update dependecies

felixDev22 committed:
Add navbar, install fontawesome

felixDev22 committed:
Add Navbar styling

felixDev22 committed:
Add images, Create home page structure

felixDev22 committed:
Add Details look and feel layout

felixDev22 committed:
Adjust card color

felixDev22 committed:
Remove navbar padding

felixDev22 committed:
Merge pull request #1 from felixDev22/api-app-structure

Remove CRA boiler plate files

felixDev22 committed:
Populate homepage data to the UI, Add styling

felixDev22 committed:
Add country card, styling and upadte dependencies

felixDev22 committed:
Update App and weatherApi

felixDev22 committed:
Update weather Api, use lat and lon

felixDev22 committed:
Add color to countries - hero section

felixDev22 committed:
Update details page, display list of items on UI

felixDev22 committed:
Fix linters error

felixDev22 committed:
Merge pull request #2 from felixDev22/display-list

Display list

felixDev22 committed:
Add App test and reducer test, update dependencies

felixDev22 committed:
Add new image for hero section

felixDev22 committed:
Add styling for home and detail page

felixDev22 committed:
Merge pull request #3 from felixDev22/testing

Testing

felixDev22 committed:
Fine-tune details page

felixDev22 committed:
Update readme

felixDev22 committed:
Merge pull request #4 from felixDev22/deploy

Deploy

felixDev22 committed:
Add live link an loom link

felixDev22 committed:
Merge pull request #5 from felixDev22/live-link

Add live link an loom link

Qoosim commented on src/__test__/App.test.js:
```diff
@@ -0,0 +1,51 @@
+
+test('should render the Homepage', () => {
+  Homepage.mockImplementation(() => <h2>Welcome From Home Page</h2>);
+  render(
+    <MemoryRouter initialEntries={['/']}>
+      <Provider store={store}>
+        <App />
+      </Provider>
+    </MemoryRouter>,
+  );
+  const txt = screen.getByText('Welcome From Home Page');
+  expect(txt).toBeInTheDocument();
+});
+
+test('should render the Details Page', () => {
+  Details.mockImplementation(() => <h2>Angola From Details Page</h2>);
+  render(
+    <MemoryRouter initialEntries={['/details/Angola']}>
+      <Provider store={store}>
+        <App />
+      </Provider>
+    </MemoryRouter>,
+  );
+  const txt = screen.getByText('Angola From Details Page');
+  expect(txt).toBeInTheDocument();
+});
+
+describe('Test app.js with pages', () => {
+  test('should render the Header and Layout components', () => {
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('Weather App');
+    expect(txt).toBeInTheDocument();
+  });
+});
```
- [ ] Wonderful job for this project. You really put in nice effort. However, there're issues with the test for mocking both the Homepage and Details pages. Kindly give a closer look and fix where the issues are.

![test-fail-1](https://user-images.githubusercontent.com/26597432/218771690-4aee38dd-88c4-49e8-9c45-48630434f602.png)

![test-fail-2](https://user-images.githubusercontent.com/26597432/218771733-112eeab4-4a67-4d06-ad52-a0a42e06a788.png)

Qoosim reviewed (changes requested):
#  *Status: Required Changes* :recycle: 

Hi @felixDev22, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

##   Top Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 


![](https://user-images.githubusercontent.com/80612925/157397398-00904e26-60ec-46cd-9510-240661c250ef.gif)

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

felixDev22 committed:
Implementent required change, fix app.test.js

levy002 commented on src/__test__/App.test.js:
```diff
@@ -0,0 +1,52 @@
+describe('Testing my App.js with pages', () => {
+  test('should render the Header and Layout components', () => {
+    Homepage.mockImplementation(() => <h1>WELCOME TO AFRICA</h1>);
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('WELCOME TO AFRICA');
+    expect(txt).toBeInTheDocument();
+  });
+
+  test('should render the Homepage', () => {
+    Homepage.mockImplementation(() => <h1>Hello From Home Page</h1>);
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('Hello From Home Page');
+    expect(txt).toBeInTheDocument();
+  });
+
+  test('should render the Details Page', () => {
+    Details.mockImplementation(() => <h1>Hello From Details Page</h1>);
+    render(
+      <MemoryRouter initialEntries={['/details']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('Hello From Details Page');
+    expect(txt).toBeInTheDocument();
+  });
+});
```
- [ ] As requested by the previous reviewer, kindly make sure to solve the `App.test.js` because at this moment it is failing. Please fix it and make sure all tests are passing.

   
   ![image](https://user-images.githubusercontent.com/87197412/219016705-24201b65-e02e-48c7-952c-95ddb7e6aa5b.png)

   ![image](https://user-images.githubusercontent.com/87197412/219016803-3bd56862-b9d9-4bd4-9bf7-34640fd6afcf.png)

   ![image](https://user-images.githubusercontent.com/87197412/219017073-730e2d9f-9b81-4dde-87f5-2a96d79c4a6e.png)

   ![image](https://user-images.githubusercontent.com/87197412/219017147-e25be12b-4b88-48ff-9d23-3dba5fe365f1.png)

levy002 reviewed (changes requested):
## Hello @felixDev22  , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

![](https://media.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif)

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

felixDev22 commented:
Thank you ...Noted..I will fix the error

felixDev22 committed:
Fix App.test.js, made sure all test pass

felixDev22 commented:
Hi 👋    Thank you for your feedback, I have implemented the following:

- [ ] I have fixed the error on App.test.js.
- [ ] I have made sure that all the test pass

As is shown below 😃 

![Screenshot 2023-02-15 at 4 56 42 PM](https://user-images.githubusercontent.com/106531695/219049940-080414e0-5881-4a64-b5fe-1551ae702f62.png)

gemmen29 reviewed (approved):
Hi @felixDev22, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

![](https://media.giphy.com/media/mCIjCgs3nWQWfJZvPA/giphy-downsized-large.gif)

## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

felixDev22 commented:
Thank you. 👍  😄
